src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -351,8 +351,24 @@ public static bool TryText(FormattingInfo formatInfo, IRContext irContext, Formu
                 return false;
             }
 
-            Contract.Assert(StringValue.AllowedListConvertToString.Contains(value.Type));
-
+            Contract.Assert(StringValue.AllowedListConvertToString.Contains(value.Type));
+
+            bool hasFormatCulture = false;
+            string enUsCultureName = "en-US";
+            CultureInfo formatCulture = culture;
+            string revFormatStr = formatString;
+            if (!string.IsNullOrEmpty(formatString) && formatString.Contains("[$-"))
+            {
+                hasFormatCulture = true;
+                int startIdx = formatString.IndexOf("[$-", StringComparison.Ordinal) + 3;
+                int endIdx = formatString.IndexOf("]", StringComparison.Ordinal);
+                formatCulture = CultureInfo.GetCultureInfo(formatString.Substring(startIdx, endIdx - startIdx));
+                formatString = formatString.Substring(endIdx + 1);
+                revFormatStr = formatString.Replace(",", "!");
+                revFormatStr = revFormatStr.Replace(".", ",");
+                revFormatStr = revFormatStr.Replace("!", ".");

```

Comment:
There are cases other than `.` and `,` (and vice-versa) for decimal separator and thousands grouping. For example, `Text(1234.5678, "[$-fr-FR]# ##0,00", "en-US")` returns "1,234.57" (the thousands grouping in fr-FR is a space)

Reply:
Add space handling here.